services/workers/src/test/java/org/sagebionetworks/table/worker/TableCSVDownloadWorkerIntegrationTest.java

```diff
@@ -435,6 +457,42 @@ List<String[]> downloadCSV(UserInfo user, DownloadFromTableRequest request) thro
 			temp.delete();
 		}
 	}
+
+	List<String[]> downloadCSVWithCustomFileName(UserInfo user, DownloadFromTableRequest request) throws Throwable {
+		// submit the job
+		AsynchronousJobStatus status = asynchJobStatusManager.startJob(user, request);
+		// Wait for the job to complete.
+		status = waitForStatus(status);
+		assertNotNull(status);
+		assertNotNull(status.getResponseBody());
+		assertTrue(status.getResponseBody() instanceof DownloadFromTableResult);
+		DownloadFromTableResult response = (DownloadFromTableResult) status.getResponseBody();
+		assertNotNull(response.getEtag());
+		assertNotNull(response.getResultsFileHandleId());
+		// Get the filehandle
+		fileHandle = (S3FileHandle) fileHandleDao.get(response.getResultsFileHandleId());
+		// Read the CSV
+		CSVReader csvReader;
+		assertEquals("text/csv", fileHandle.getContentType());
+		assertEquals("file1234.csv", fileHandle.getFileName());

```

Comment:
Let us just add this assert with the request.getFileName() in the method we already have and remove this one.